tests/test_polydata.py

```diff
@@ -222,6 +222,12 @@ def test_geodesic_distance(sphere):
     distance = sphere.geodesic_distance(0, sphere.n_points - 1)
     assert isinstance(distance, float)
 
+    # Use scalar weights
+    distance_use_scalar_weights = sphere.geodesic_distance(
+        0, sphere.n_points - 1, use_scalar_weights=True
+    )
+    assert isinstance(distance_use_scalar_weights, float)
+
```

Comment:
Judging by CI this seems to segfault on VTK 8, might need a conditional skip.

Reply:
>  I get a segfault whether or not the scalars are point or cell data

As a side note, looking at the [vtk code](https://github.com/Kitware/VTK/blob/master/Filters/Modeling/vtkDijkstraGraphGeodesicPath.cxx#L158-L176) it seems it will only work with point data. Maybe worth mentioning this in the docstring?

Reply:
Yeah, if that's the case then we should definitely mention that, maybe even raise if there aren't any point scalars. We have a few similar filters lying around in the codebase.

Reply:
Fix by d34157a801f206ed818698b40d2fcdac665c8800 .

Reply:
I updated the docstring to include a not about VTK>=9 and point data. Further, I make sure the filter itself skips this option if not VTK9